components/AmountInputCard.js

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from 'react';
+import { Grid, Card, Input, Link } from '@makerdao/ui-components-core';
+import { TextBlock } from './Typography';
+
+export default function({
+  title,
+  children,
+
+  // a Currency class like SAI or DAI
+  unit,
+
+  // the maximum allowable value, as an instance of `unit`
+  max,
+
+  // a function that gets one argument: the input, typed to the `unit` class. it
+  // should return an error message if the value is invalid, and nothing if it's
+  // valid
+  validate,
+
+  // this will be called with the final typed value if validation passes
+  update
+}) {
+  const [error, setError] = useState();
+  const [value, setValue] = useState('');
+
+  const setMax = () => {
+    setValue(max.toBigNumber().toString());
+    update(max);
+  };
+
+  const onChange = event => {
+    const { value } = event.target;
+    setValue(value);
+
+    if (isNaN(parseFloat(value)))
```

Comment:
This is a "just-in-case" check, since modern browsers should respect the `type="number"` on the input tag and disallow non-numeric input